.github/workflows/lint.yml

```diff
@@ -24,3 +24,22 @@ jobs:
         with:
           python-version: "3.x"
       - uses: pre-commit/action@v3.0.0
+
+  patchcheck:
+    if: "github.repository == 'python/cpython' && github.event_name == 'pull_request'"
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          ref: ${{ github.event.pull_request.head.ref }}
```

Comment:
The job fails: need a git checkout to get modified files
Getting base branch for PR ... not a PR branch

Reply:
I finally found what's wrong:

```
 Getting base branch for PR ... No /opt/hostedtoolcache/Python/3.11.5/x64/lib/python3.11/config-3.11-x86_64-linux-gnu
not a PR branch
```

`SRCDIR = sysconfig.get_config_var('srcdir')` does not have `.git` directory, that's why everything fails :)